test/libsais_memory_budget_test.sh

```
@@ -0,0 +1,73 @@
+ROOT="$(cd "$HERE/.." && pwd)"
+BWAMEM2="$ROOT/bwa-mem2"
+FA_SRC="$HERE/fixtures/synthetic_1mb.fa"
+[[ -s "$FA_SRC" ]] || { echo "FAIL: $FA_SRC missing"; exit 1; }
+
+UNAME_S="$(uname -s)"
+if [[ "$UNAME_S" == "Darwin" ]]; then
+    TIME_CMD=(/usr/bin/time -l)
+    PEAK_RE='^ *([0-9]+) +maximum resident set size'
+else
+    TIME_CMD=(/usr/bin/time -v)
+    PEAK_RE='Maximum resident set size \(kbytes\): +([0-9]+)'
```

Shellcheck (0.11.0)

[warning] PEAK_RE appears unused. Verify use (or export if used externally).

(SC2034)